acrg/BC/BC_CAMS.py

```diff
@@ -54,70 +54,93 @@
 
 acrg_path = Paths.acrg
 data_path = Paths.data
-cams_directory = os.path.join(data_path, 'ECMWF_CAMS', 'CAMS_inversion/')
```

Comment:
readCAMSinversion defaults to the path here for cams_directory. The reason I've given the input as None rather than the path is because then the function can be called with cams_directory=None or with cams_directory='some_path'
e.g. on lines 520-523 where we call domain_volume we need to use an if statement to call it with different inputs depending on the input given for fp_directory in makeCAMSBC, whereas if the fp_directory input in domain_volume was None with a default set in the function then we wouldn't need the if statement

Reply:
Well, the `fp_directory` clause is kind of a special case and is related to how we had to set up the config (ini) files. Basically we initially had to specify something and so None became a sensible de facto for using the default. Largely if you can get away with not passing anything and using the defaults that makes more sense e.g. you could use a **kwargs argument with a dictionary and omit any variables which weren't needed.

Reply:
That makes sense, although I do think it is more flexible and more readable when you call the function if there is the option to give None for input variables and then then function uses the default
Good idea to use kwargs instead of an if statement

Reply:
The Python default argument syntax is intended to be used exactly for the case where you want default values to be provided and without the user needing to specify them. There are special cases (e.g. mutable default arguments) where you need to use =None format but those are quite specific and wouldn't apply here. If going against the convention, largely there should be a reason and a benefit (e.g. is there an example of when you'd want to call `version` with `None`?). It's not as applicable now, but though this perhaps provides some more flexibility I don't think it makes the function definition (or call) more readable (and may in fact impact maintainability).

Reply:
Okay, I have taken the option out for version and can do for cams_directory too if that goes best with what how we want the code to be written.
Personally I find it easiest to have the option to give None in some cases e.g. in the case I gave earlier where I find it more awkward (and therefore I think the code reads more awkwardly) to have to find a work around like we have there, but if that's not best practise then that's fine and I'll change it

Reply:
I'm not saying to remove the option for version etc. if there's a use case for it - it sounds like you do have one, so it would useful if you could provide it? I'm just a bit wary of defining things twice and have caused myself problems in the past by trying to be too kind and flexible.